### PR TITLE
convert the wrapper script to work with Hisat2-Cuffcompare-Cuffmerge.…

### PR DESCRIPTION
**Pull request by slr71**
…sh and to handle recursive uploads correctly

### CODE REVIEW

File: upload-files [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        if stat.S_ISDIR(mode):
+            upload_directory(src, dest, ticket, owner, uploader)
+        elif stat.S_ISREG(mode):
+            upload_file(src, dest, ticket, owner, uploader)
+        else:
+            print >> sys.stderr, "not uploading {0}: unsupported file type".format(src)
+
+if __name__ == "__main__":
+    args = parse_args()
+    do_upload(args.sources, args.dest, args.ticket, args.owner, args.uploader)
```
Comment (slr71) [inline upload-files:99]: The upload script has been tested with Nate Miller's phytoshell tool.

File: wrapper [hunk 2]
```diff
@@ -13,11 +13,12 @@ import subprocess
 import sys
 import re
 
-from get_gene_length_filter import get_gene_lengths
-
```
Comment (slr71) [inline wrapper:17]: It seems as though `get_gene_lengths` doesn't apply to this docker image.

File: wrapper [hunk 3]
```diff
@@ -13,11 +13,12 @@ import subprocess
...
 # The name of the configuration file.
 config_file = "config.json"
 
+# The name of the job submission file.
+job_file = "job"
+
```
Comment (slr71) [inline wrapper:21]: The job submission file has a list of jobs files to exclude from uploads.

File: wrapper [hunk 4]
```diff
@@ -106,60 +107,45 @@ def download_file(ticket, src):
 
 # Download a set of files referenced in a ticket list file from iRODS, returning a list of downloaded files.
 def download_files(ticket_list_path):
+    downloaded_files = [];
     with TicketListReader(ticket_list_path) as tlr:
         for ticket, src in tlr:
             download_file(ticket, src)
+            downloaded_files += [os.path.basname(src)]
+    return downloaded_files
```
Comment (slr71) [inline wrapper:115]: This function now returns a list of downloaded file names.

File: wrapper [hunk 5]
```diff
@@ -106,60 +107,45 @@ def download_file(ticket, src):
...
-    fullpath = os.path.join(dest, src)
-    rc = subprocess.call(["ichmod", "own", owner, fullpath])
-    if rc != 0:
-        raise Exception("could not change the owner of {0} to {1}".format(fullpath, owner))
-    rc = subprocess.call(["ichmod", "null", irods_user, fullpath])
-    if rc != 0:
-        raise Exception("could not remove {0} permissions on {1}".format(irods_user, fullpath))
+        rc = subprocess.call(["Hisat2-Cuffcompare-Cuffmerge.sh"] + arguments, stdout=out, stderr=err)
+        if rc != 0:
+            raise Exception("Hisat2-Cuffcompare-Cuffmerge.sh returned exit code {0}".format(rc))
```
Comment (slr71) [inline wrapper:122]: `run_job` can be much simpler for this version of the script. `upload_file` has been replaced by the separate script.

File: wrapper [hunk 6]
```diff
@@ -188,19 +175,18 @@ if __name__ == "__main__":
...
     except Exception as e:
         updater.running("job encountered an error: {0}".format(e))
         job_failed = True
 
     # Upload the output files.
     try:
         updater.running("uploading the output files")
-        files_to_upload = [cfg.stdout, cfg.stderr] + output_files
-        upload_files(cfg.output_ticket_list, cfg.irods_user, cfg.irods_job_user, files_to_upload, updater)
+        paths = get_paths_to_upload(get_excluded_files(cfg))
```
Comment (slr71) [inline wrapper]: I just realized that I forgot to include the input files in the list of files to exclude. I'll push a change to fix that momentarily.

File: upload-files (hunk 1 above)
Comment (upendrak) [reply upload-files:99]: Got it..

File: wrapper (hunk 2 above)
Comment (upendrak) [reply wrapper:17]: You are absolutely right. Forgot to remove that

**Pull request merged**